explore_utils.py

```diff
@@ -20,6 +20,48 @@ def find_most_visit(dataset):
     return max_visit
 
 
+def find_fraction_of_customers_with_non_zero_revenue(dataset):
+    """Find the fraction of customers in the google dataset with non-zero revenue.(only in train set)
+    
+    args:
+        dataset (Dataset): the google analytics dataset.
+        
+    returns:
+            The fraction of customers with non-zero revenue.
+    """
+    
+    
+    train_df = dataset.train
+    train_df['revenue'] = train_df['totals.transactionRevenue'].astype(float)
+    
+    """
+        # This is a counter variable that increases for each customer with non-zero revenue.
+        customer_zero_revenue_count = 0
+    
+        # for x in the number of rows (customers) of train_df
+        for x in train_df:
+            if (train_df['revenue'] != 0):
+                customer_zero_revenue_count = customer_zero_revenue_count + 1
+    
+        total_num_customers = len(train_df)
+    
+        final_fraction = customer_zero_revenue_count / total_num_customers
+    
+        return final_fraction
+    """
+    
+    # The number of customers who have non-zero revenue.
+    num_customers_non_zero = np.count_nonzero(~np.isnan(train_df['revenue']))
```

Comment:
Since you don't group by visitor id, this is actually the number of non-zero revenue _transactions_, not customers, since customers can have multiple transactions. Both quantities are interesting to know, but currently your result doesn't match your description.